CMakeLists.txt

```diff
@@ -2,17 +2,21 @@ cmake_minimum_required(VERSION 3.5)
 project(xacro)
 
 find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_python REQUIRED)
 
 install(FILES scripts/completion.bash DESTINATION share/${PROJECT_NAME}/environment)
 
 ## The following manually installs the python package and the entry-point script
 ## Using ament_cmake, we cannot actually use setup.py to install these...
 
-# install xacro python package
-ament_python_install_package(xacro)
-# install main script
-install(PROGRAMS scripts/xacro DESTINATION bin)
-install(PROGRAMS scripts/xacro DESTINATION lib/${PROJECT_NAME})
+# install xacro python package w/ console_script entrypoint
+ament_python_install_package(${PROJECT_NAME}
+    SCRIPTS_DESTINATION lib/${PROJECT_NAME})
+
+# copy generated script executables to bin dir so it can be run without ros2 run
+install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_python/${PROJECT_NAME}/scripts/
+    DESTINATION bin/)
```

Comment:
`ament_python_install_package` now installs the console_script entrypoint so that xacro can be run using `ros2 run xacro xacro`. these lines here make it so that people can still directly call `xacro` since bin is on the PATH but lib isnt